test/functional/pt_osc_migration_functional_test.rb

```diff
@@ -38,6 +38,22 @@ class TestMigration < ActiveRecord::PtOscMigration; end
           ActiveRecord::Base.connection.execute "DROP TABLE IF EXISTS `#{@table_name}`;"
         end
 
+        context 'a raw SQL execute migration' do
+          setup do
+            TestMigration.class_eval <<-EVAL
+            def up
+              execute 'SELECT 1'
+            end
+            EVAL
+          end
+
+          should 'throw an exception' do
+            assert_raise ActiveRecord::UnsupportedMigrationError do
+              @migration.migrate(:up)
+            end
+          end
```

Comment:
A critical piece of this is that the `pt-osc` command never gets run as it'll create a non-functioning migration. That should be tested for — perhaps asserting that `execute_pt_osc` and `print_pt_osc` are never called.